### PR TITLE
cilium: fixes transparent encryption

### PR DESCRIPTION
**Pull request by jrfastab**
Fix a couple issues with transparent encryption. First ensure encryption mark field is not replaced with source identity. If packet is being encrypted it is a remote packet and source identity in the mark field is not needed. Next on kernels without FIB support we may still have a encryption interface. By using the interface with redirect encryption will continue to work even when an explicit route has not been configured. With above two fixes GKE using standard configuration from guides will work correctly.

Finally, on GKE systems and a local cluster the endpoint k8s watcher was reporting a zero key. By doing an explicit assignment this is resolved. I'm not entirely clear what is happening here but its clear the event has a correct encryption key identifier and the post translated even drops the encryption key id. By doing the assignment directly the key is now correct. TBD understand golang parts here.

```release-note
fix transparent encryption related bugs
```

### CODE REVIEW

**Comment by jrfastab** [comment]
test-focus K8sDatapathConfig.*Check connectivity with transparent encryption"

File: pkg/k8s/factory_functions.go [hunk 1]
```diff
@@ -684,7 +686,7 @@ func ConvertToCiliumEndpoint(obj interface{}) interface{} {
 				Labels:      nil,
 				Annotations: nil,
 			},
-			Encryption: &concreteObj.Status.Encryption,
+			Encryption: &encryptVal,
```
Comment (aanm) [inline pkg/k8s/factory_functions.go]: Is this change really needed?

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/k8s/factory_functions.go (hunk 1 above)
Comment (jrfastab) [reply pkg/k8s/factory_functions.go]: I don't understand it at the moment but its causing encryption key to be zero without it on my GKE cluster.

**Comment by aanm** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Review by borkmann — APPROVED** [review]

**Comment by jrfastab** [comment]
Issue filed for K8s-1.11-Kernel-netnext, that appears to be unrelated those tests are running without encryption.

https://github.com/cilium/cilium/issues/11985

**Comment by jrfastab** [comment]
retest-net-next

**Review by jrfastab — COMMENTED** [review]
LGTM thanks

**Review by joestringer — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Review by jrfastab — COMMENTED** [review]
LGTM thanks!
